emoji/src/main/java/com/vanniktech/emoji/EmojiPagerAdapter.java

```diff
@@ -27,23 +30,26 @@
 
   @Override public Object instantiateItem(final ViewGroup pager, final int position) {
     final View newView;
-
     if (position == 0) {
       newView = new RecentEmojiGridView(pager.getContext()).init(listener, longListener, recentEmoji);
-
       recentEmojiGridView = (RecentEmojiGridView) newView;
     } else {
       newView = new EmojiGridView(pager.getContext()).init(listener, longListener, EmojiManager.getInstance().getCategories()[position - 1]);
+      gridViewsList.add((EmojiGridView) newView);
     }
-
     pager.addView(newView);
-
     return newView;
   }
 
+  public void notifyPagesChanged(){
```

Comment:
this is probably causing the lag. Also it's not optimal. How about just changing the page that needs to be changed? (it's the one you're in and the recent one.